From 4b446af29a458aa03165ee9dac91a540c1cc32d7 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:59 -0400
Subject: [powerpc] Fix location and rename exception trampolines

Message-id: <1410545655-205645-71-git-send-email-dzickus@redhat.com>
Patchwork-id: 94156
O-Subject: [RHEL7 PATCH 070/626] powerpc: Fix location and rename exception trampolines
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit fa111f1f764783fd5f1f12f5dd5d5f66d3160b48
Author: Michael Ellerman <michael@ellerman.id.au>
Date:   Thu Aug 15 15:22:17 2013 +1000

    powerpc: Fix location and rename exception trampolines

    The symbols that name some of our exception trampolines are ahead of the
    location they name. In most cases this is OK because the code is tightly
    packed, but in some cases it means the symbol floats ahead of the
    correct location, eg:

      c000000000000ea0 <performance_monitor_pSeries_1>:
              ...
      c000000000000f00:       7d b2 43 a6     mtsprg  2,r13

    Fix them all by moving the symbol after the set of the location.

    While we're moving them anyway, rename them to loose the camelcase and
    to make it clear that they are trampolines.

    Signed-off-by: Michael Ellerman <michael@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/exceptions-64s.S b/arch/powerpc/kernel/exceptions-64s.S
index a44ce5d..d805af6 100644
--- a/arch/powerpc/kernel/exceptions-64s.S
+++ b/arch/powerpc/kernel/exceptions-64s.S
@@ -323,32 +323,32 @@ hv_exception_trampoline:
   * prolog code of the PerformanceMonitor one. A little
   * trickery is thus necessary
   */
-performance_monitor_pSeries_1:
  . = 0xf00
+performance_monitor_pseries_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b performance_monitor_pSeries
 
-altivec_unavailable_pSeries_1:
  . = 0xf20
+altivec_unavailable_pseries_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b altivec_unavailable_pSeries
 
-vsx_unavailable_pSeries_1:
  . = 0xf40
+vsx_unavailable_pseries_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b vsx_unavailable_pSeries
 
-facility_unavailable_trampoline:
  . = 0xf60
+facility_unavailable_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b facility_unavailable_pSeries
 
-hv_facility_unavailable_trampoline:
  . = 0xf80
+hv_facility_unavailable_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b facility_unavailable_hv
@@ -821,32 +821,32 @@ system_call_relon_pSeries:
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b h_doorbell_relon_hv
 
-performance_monitor_relon_pSeries_1:
  . = 0x4f00
+performance_monitor_relon_pseries_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b performance_monitor_relon_pSeries
 
-altivec_unavailable_relon_pSeries_1:
  . = 0x4f20
+altivec_unavailable_relon_pseries_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b altivec_unavailable_relon_pSeries
 
-vsx_unavailable_relon_pSeries_1:
  . = 0x4f40
+vsx_unavailable_relon_pseries_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b vsx_unavailable_relon_pSeries
 
-facility_unavailable_relon_trampoline:
  . = 0x4f60
+facility_unavailable_relon_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b facility_unavailable_relon_pSeries
 
-hv_facility_unavailable_relon_trampoline:
  . = 0x4f80
+hv_facility_unavailable_relon_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b hv_facility_unavailable_relon_hv
-- 
1.7.1